uwnet/datasets.py

```diff
@@ -128,6 +128,50 @@ def scale(self):
         return valmap(lambda x: x.max(), std)
 
 
+class XRTimeSeriesEta(XRTimeSeries):
+    """Same as XRTimeSeries, but only trained on a specific eta value.
+    """
+
+    def __init__(self, data, eta):
+        """
+        Parameters
+        ----------
+        data : xr.DataArray
+            An input dataset. This dataset must contain at least some variables
+            with all of the dimensions ['time' , 'z', 'x', 'y'].
+        eta : int
+            The value of eta to train on.
+        """
+        self.eta = eta
+        self.data = data
+        self.time_length = 2
+        self.numpy_data = {key: data[key].values for key in data.data_vars}
+        self.data_vars = set(data.data_vars)
+        self.dims = {key: data[key].dims for key in data.data_vars}
+        self.constants = {
+            key
+            for key in data.data_vars
+            if len({'x', 'y', 'time'} & set(data[key].dims)) == 0
+        }
+        self.setup_indices()
+
+    def setup_indices(self):
+        len_x = len(self.data['x'].values)
+        len_y = len(self.data['y'].values)
+        len_t = len(self.data['time'].values)
+
+        x_iter = range(0, len_x, 1)
+        y_iter = range(0, len_y, 1)
+        t_iter = range(0, len_t, 2)
```

Comment:
I think this code will skip every other time point, when what we want is to yield all possible (time, time+1) pairs.  Let's replace this with
```
t_iter = range(0, len_t-1, 1)
```

Reply:
Good point!